.circleci/config.yml

```diff
@@ -31,15 +29,60 @@ jobs:
       - run:
           name: Start LDAP
           command: sudo /etc/init.d/slapd stop
-      - run:
-          name: save SHA to a file
-          command: echo $CIRCLE_SHA1 > .circle-sha
       - restore_cache:
           keys:
-            - v1-repo-{{ checksum ".circle-sha" }}
+            - v1-repo-{{ .Environment.CIRCLE_SHA1 }}
       - run: go get -u github.com/golang/lint/golint
       - run: make testall
 
+  build_binary:
+    docker:
+      - image: circleci/golang:1.8
+    working_directory: /go/src/github.com/coreos/dex
+    steps:
+      - restore_cache:
+          keys:
+            - v1-repo-{{ .Environment.CIRCLE_SHA1 }}
+      - run:
+          name: Build dex binary
+          command: |
+            make release-binary
+            mkdir -p _output/bin/dex && cp /go/bin/dex _output/bin/dex
+      - save_cache:
+          key: v1-binary-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - /go/src/github.com/coreos/dex/_output/bin
+
+  deploy_image:
+    docker:
+      - image: circleci/golang:1.8
```

Comment:
So I tried to use the official `docker` container, and the problem was it didn't have a bunch of circleci dependencies installed like `tar` `gzip` and whatnot, so we needed a separate step to install things anyways, and I figured installing those from a package manager was probably going to take more time than downloading this one docker binary, so I reverted it.